conanfile.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os.path
+from conans import ConanFile, CMake
+from conans.tools import Git
+from conans.errors import ConanInvalidConfiguration
+
+class ImguiSfmlConan(ConanFile):
+    """ Imgui-SFML library conan file.
+
+    Options:
+
+    * shared: True or False
+      Whether to build this library shared or static
+
+    * fPIC: True or False
+      Whether or not to use -fPIC option while building
+
+    * imconfig: None or String
+      Use 'None' if you want this library to provide default
+      ImGui's imconfig.h file. Otherwise, provide a path to the
+      imconfig.h file you want to include in the build.
+
+    * imconfig_install_folder: None or String
+      Use 'None' if you want this build to include imconfig.h file,
+      otherwise use anything else.
+
+    * imgui_revision: String
+      Tag or branch of ImGui repository that should be used with
+      ImGui-SFML.
+    """
+
+    name = 'imgui-sfml'
+    version = '2.0.1'
+    description = 'ImGui binding for use with SFML'
+    topics = ('conan', 'sfml', 'gui', 'imgui')
+    url = 'https://github.com/eliasdaler/imgui-sfml'
+    homepage = 'https://github.com/eliasdaler/imgui-sfml'
+    author = 'Elias Daler <eliasdaler@yandex.ru>'
+    requires = 'sfml/2.5.1@bincrafters/stable'
+    license = 'MIT'
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'imconfig': 'ANY',
+        'imconfig_install_folder': 'ANY',
+        'imgui_revision': 'ANY'
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'imconfig': None,
+        'imconfig_install_folder': None,
+        'imgui_revision': 'v1.70'
+    }
+    exports_sources = [
+        'cmake/FindImGui.cmake'
+        'CMakeLists.txt',
+        'imconfig-SFML.h',
+        'imgui-SFML.cpp',
+        'imgui-SFML_export.h',
+        'imgui-SFML.h',
+    ]
+    exports = 'LICENSE.md'
+    _imgui_dir = 'imgui'
+
+    def source(self):
+        git = Git(folder=self._imgui_dir)
+        git.clone('https://github.com/ocornut/imgui.git', str(self.options.imgui_revision))
+
+    def configure(self):
+        imconfig = self.options.imconfig
+        if imconfig and not os.path.isfile(str(imconfig)):
+            raise ConanInvalidConfiguration("Provided ImGui config is not a file or doesn't exist")
+        if not self.options.imgui_revision:
+            raise ConanInvalidConfiguration("ImGui revision is empty. Try latest version tag or 'master'")
+
+        self.options['sfml'].graphics = True
+        self.options['sfml'].window = True
+        self.options['sfml'].shared = self.options.shared
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions['IMGUI_DIR'] = os.path.join(self.source_folder, self._imgui_dir)
+        cmake.definitions['SFML_DIR'] = os.path.join(self.deps_cpp_info['sfml'].lib_paths[0], 'cmake', 'SFML')
```

Comment:
Need to set `SFML_STATIC_LIBRARIES` if self.options.shared is `False`.

Reply:
Actually, this raises the question - is there any way to SFML Conan package maintainers to specify that SFML_STATIC_LIBRARIES need to be defined when trying to link with static SFML libs?

Reply:
> Need to set `SFML_STATIC_LIBRARIES`

I didn't find any `SFML_STATIC_LIBRARIES` variables in SFML CMake nor in ImGui-SFML CMake. Is it something internal within CMake itself ?

Also, setting `self.options['sfml'].shared = ...` should resolve anything related to sfml and shared/static. Because otherwise it will be some kind of workaround.

> is there any way to SFML Conan package maintainers to specify that SFML_STATIC_LIBRARIES need to be defined when trying to link with static SFML libs?

I believe, they [already doing](https://github.com/bincrafters/conan-sfml/blob/testing/2.5.1/conanfile.py#L140) something like that.

Reply:
'SFML_STATIC' is a preprocessor define. 'SFML_STATIC_LIBRARIES' is a CMake define and tells CMake that it needs to search for static SFML libraries.
Without it, I can't build ImGui-SFML as static lib, because it fails to find static SFML libs. See this:
https://github.com/SFML/SFML/blob/ff011dc51dc47373b0d49b087f8ac9d3d2bc8b06/cmake/SFMLConfig.cmake.in#L19

As an alternative (and I think it'll be better for ImGui-SFML) - let's set (SFML_STATIC_LIBRARIES as ON in ImGui's SFML's CMake is BUILD_SHARED_LIBS is set)

```cmake
if (BUILD_SHARED_LIBS)
    set(SFML_STATIC_LIBRARIES CACHE INTERNAL ON)
endif()
```

^ with this in ImGui-SFML's cmake, we won't need to set it in Conan recipe.

Reply:
Ah, I see. That's why I couldn't build it. 

As for the solution, I think it's it's more of a general build issue, so it's right to set this define in CMake in case of somebody building the library manually !

Reply:
Indeed. I think it's worth doing it in this PR.